src/object.js

```diff
@@ -163,7 +180,7 @@ function defineProp(object, name, opts = {}) {
   });
 
   Object.defineProperty(object, name, {
-    get: function() { return this._getProp(name); },
+    get: createPropGetter(descriptor, name),
```

Comment:
Could you accomplish the same thing by overriding `#_getProp` on `Model` instead? That would remove the hacky `descriptor.attr` check here.